ethereum/spec/src/main/java/tech/pegasys/teku/spec/executionlayer/ExecutionLayerChannelStub.java

```diff
@@ -348,6 +351,22 @@ public SafeFuture<List<ClientVersion>> engineGetClientVersion(final ClientVersio
     return SafeFuture.completedFuture(List.of(STUB_CLIENT_VERSION));
   }
 
+  @Override
+  public SafeFuture<List<BlobAndProof>> engineGetBlobs(
+      final List<VersionedHash> blobVersionedHashes, final UInt64 slot) {
+    final List<Blob> blobs =
+        blobsUtil.generateBlobs(
+            slot,
+            blobsToGenerate.orElseGet(
+                () -> random.nextInt(spec.getMaxBlobsPerBlock().orElseThrow() + 1)));
+    final List<KZGCommitment> commitments = blobsUtil.blobsToKzgCommitments(blobs);
+    final List<KZGProof> proofs = blobsUtil.computeKzgProofs(blobs, commitments);
+    return SafeFuture.completedFuture(
+        IntStream.range(0, blobs.size())
+            .mapToObj(index -> new BlobAndProof(blobs.get(index), proofs.get(index)))
+            .toList());
+  }
```

Comment:
we need to be careful returning random stuff here because we could make our production code go in an inconsistent state and stop working with the stub.
Maybe it is not a problem but currently my gut says we should think about it.

Reply:
I'm thinking we could either return an empty list and consider the `engineGetBlobs` unavailable when using the stub EL or simulate the blobs mempool by caching the generated blobs when we call `engineGetPayload` for the block generation and serve those (a LRU cache for example). Using a cache might make the blobs always available from the mempool though unless adjusting the cache capacity to make them unavailable sometimes

Reply:
better no data, for sure

Reply:
To avoid nulls we could migrate to `List<Optional<BlobAndProof>>`